### PR TITLE
Fix flaky test in TomcatServiceDestroyedConnectorTest

### LINKED ISSUE #3247: Test failure: `TomcatServiceDestroyedConnectorTest.serviceUnavailableAfterConnectorIsDestroyed()`
```
TomcatServiceDestroyedConnectorTest > serviceUnavailableAfterConnectorIsDestroyed() FAILED
    java.lang.AssertionError: 
    Expecting:
     <200 OK>
    and actual:
     <503 Service Unavailable>
    to refer to the same object
        at com.linecorp.armeria.server.tomcat.TomcatServiceDestroyedConnectorTest.serviceUnavailableAfterConnectorIsDestroyed(TomcatServiceDestroyedConnectorTest.java:55)
```

### PR DESCRIPTION
**Pull request by minwoox**
Close #3247

### CODE REVIEW

File: tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java [hunk 1]
```diff
@@ -46,16 +47,21 @@ protected void configure(ServerBuilder sb) throws Exception {
...
         }
+
+        @Override
+        protected boolean runForEachTest() {
+            return true;
+        }
     };
 
-    @Test
+    @RepeatedTest(100)
```
Comment (trustin) [inline tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java:57]: Revert 😄

Comment (minwoox) [reply tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceDestroyedConnectorTest.java:57]: Ah, I'm testing to check if it happens again. 😄

**Comment by minwoox** [comment]
The `LifecycleState` is changed to `NEW` after destroyed. That was the reason. 😄

**Review by trustin — APPROVED** [review]
Thanks, @minwoox :bow:

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox

**Pull request merged**